### PR TITLE
Add OutputGroupInfo for pkg_rpm rule

### PR DESCRIPTION
**Pull request by k0walik**
The `pkg_deb` rule returns OutputGroupInfo that contains `deb = [output_file]` output group. We could add similar return value for `pkg_rpm` rule.

Here is the implementation for `pkg_deb`: https://github.com/bazelbuild/rules_pkg/blob/main/pkg/private/deb/deb.bzl#L166

The open question is whether we should match other groups? `changes` could contain `ctx.attr.changelog` value, but I'm not quite sure what would fit for `out` group.

### CODE REVIEW

**Comment by aiuto** [comment]
But why?. Is there a bug that describes the thing you are trying to
accomplish.

**Comment by k0walik** [comment]
Actually, I think the problem is not the OutputGroupInfo part, but DefaultInfo mismatch between `pkg_deb` and `pkg_rpm`:
- `pkg_deb` returns `files = depset([output_file])`, which is a single file: https://github.com/bazelbuild/rules_pkg/blob/main/pkg/private/deb/deb.bzl#L168
- `pkg_rpm` returns `files = depset([outputs])`, which contains two files: https://github.com/bazelbuild/rules_pkg/blob/main/pkg/rpm_pfg.bzl#L664

File: tests/deb_rpm_tar/BUILD [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    name = "test_deb",
+    data = ":deb_input",
+    description = "Some package",
+    maintainer = "john.doe@example.com",
+    package = "debpackage",
+    package_file_name = "debpackage-0.0.0.deb",
+    version = "0.0.0",
+)
+
+pkg_tar(
```
Comment (aiuto) [inline tests/deb_rpm_tar/BUILD:27]: Why would someone do this?

Comment (k0walik) [reply tests/deb_rpm_tar/BUILD:27]: What if I have to ship multiple files, i.e. instructions or deployment scripts, along with deb/rpm that cannot be part of the package to the air gapped environment? The tarball is the easiest way to distribute.

File: tests/deb_rpm_tar/test.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```
Comment (aiuto) [inline tests/deb_rpm_tar/test.sh:1]: Just drop this test. If we want to test for a single file, then we should be doing an analysis test.

File: tests/deb_rpm_tar/BUILD (hunk 1 above)
Comment (aiuto) [reply tests/deb_rpm_tar/BUILD:27]: OK. 
But we don't need any of this. There are existing test targets that exhibit the right characteristics. Or, if not, they should be in the respective test/rpm or test/deb trees.

File: pkg/rpm_pfg.bzl [hunk 3]
```diff
@@ -659,7 +659,17 @@ def _pkg_rpm_impl(ctx):
...
+
+    output_groups = {
+        "out": [default_file],
+        "rpm": [output_file],
+        "changes": changes
+    }
     return [
+        OutputGroupInfo(**output_groups),
         DefaultInfo(
             files = depset(outputs),
```
Comment (aiuto) [inline pkg/rpm_pfg.bzl:674]: The right change here is to align with the other tools.  It should be
```
DefaultInfo(files = depset([output_file]), runfiles = ctx.runfiles(files = outputs))
```

BUT... that breaks existing tests.  So, what don't we reduce this CL to just the changes to this file, which are reasonable, and come back to the problem of pkg_rpm giving two output files later.

File: tests/deb_rpm_tar/test.sh (hunk 2 above)
Comment (k0walik) [reply tests/deb_rpm_tar/test.sh]: Done

**Pull request merged**